openpype/hosts/substancepainter/plugins/publish/collect_textureset_images.py

```diff
@@ -0,0 +1,196 @@
+    # TODO: Production-test usage of color spaces
+    # TODO: Detect what source data channels end up in each file
+
+    label = "Collect Texture Set images"
+    hosts = ["substancepainter"]
+    families = ["textureSet"]
+    order = pyblish.api.CollectorOrder
+
+    def process(self, instance):
+
+        config = self.get_export_config(instance)
+        asset_doc = get_asset_by_name(
+            project_name=instance.context.data["projectName"],
+            asset_name=instance.data["asset"]
+        )
+
+        instance.data["exportConfig"] = config
+        maps = get_parsed_export_maps(config)
+
+        # Let's break the instance into multiple instances to integrate
+        # a subset per generated texture or texture UDIM sequence
+        for (texture_set_name, stack_name), template_maps in maps.items():
+            self.log.info(f"Processing {texture_set_name}/{stack_name}")
+            for template, outputs in template_maps.items():
+                self.log.info(f"Processing {template}")
+                self.create_image_instance(instance, template, outputs,
+                                           asset_doc=asset_doc,
+                                           texture_set_name=texture_set_name,
+                                           stack_name=stack_name)
+
+    def create_image_instance(self, instance, template, outputs,
+                              asset_doc, texture_set_name, stack_name):
+        """Create a new instance per image or UDIM sequence.
+
+        The new instances will be of family `image`.
+
+        """
+
+        context = instance.context
+        first_filepath = outputs[0]["filepath"]
+        fnames = [os.path.basename(output["filepath"]) for output in outputs]
+        ext = os.path.splitext(first_filepath)[1]
+        assert ext.lstrip("."), f"No extension: {ext}"
+
+        always_include_texture_set_name = False  # todo: make this configurable
+        all_texture_sets = substance_painter.textureset.all_texture_sets()
+        texture_set = substance_painter.textureset.TextureSet.from_name(
+            texture_set_name
+        )
+
+        # Define the suffix we want to give this particular texture
+        # set and set up a remapped subset naming for it.
+        suffix = ""
+        if always_include_texture_set_name or len(all_texture_sets) > 1:
+            # More than one texture set, include texture set name
+            suffix += f".{texture_set_name}"
+        if texture_set.is_layered_material() and stack_name:
+            # More than one stack, include stack name
+            suffix += f".{stack_name}"
+
+        # Always include the map identifier
+        map_identifier = strip_template(template)
+        suffix += f".{map_identifier}"
+
+        image_subset = get_subset_name(
+            # TODO: The family actually isn't 'texture' currently but for now
+            #       this is only done so the subset name starts with 'texture'
```

Comment:
I'd say the only thing maybe @mkolar is this here. Any chance for subset names later will have backwards compatibility issues so wanted to note this now whether this needs changing.